### ISSUE TITLE
False positive for private call when using macroexpand hook

### ISSUE THREAD

**Issue by mrkam2** [clj-kondo/clj-kondo#2224]
**version**

```
$ clj-kondo --version
clj-kondo v2023.10.20
```

**problem**

When var quoted `#'private-fn` call to a private function is placed inside a body of a macro configured with a macroexpand hook, private-call false positive warning is triggered.

**repro**

[private-call-with-macroexpand.zip](https://github.com/clj-kondo/clj-kondo/files/13507146/private-call-with-macroexpand.zip)

`a.clj`
```clojure
(ns a)

(defn- private-fn [])

(defmacro my-macro [& body]
  `(do ~@body))
```

`b.clj`
```clojure
(ns b
  (:require
    [a :refer [my-macro]]))

(defn no-warning []
  (#'a/private-fn))

(defn produces-warning []
  (my-macro (#'a/private-fn)))
```

`.clj-kondo/config.edn`
```clojure
{:hooks {:macroexpand {a/my-macro hooks.a/my-macro}}}
```

`.clj-kondo/hooks/a.clj`
```clojure
(ns hooks.a)

(defmacro my-macro [& body]
  `(do ~@body))
```

```
$ clj-kondo --lint .
./a.clj:3:8: warning: Unused private var a/private-fn
./b.clj:9:16: error: #'a/private-fn is private
linting took 23ms, errors: 1, warnings: 1
```

**expected behavior**

No `#'a/private-fn is private` error should be shown. Note that an identical call inside the `no-warning` function triggers no such error.

**Comment by borkdude** [comment]
Thanks!

**Comment by borkdude** [comment]
This is likely because the following isn't treat the same as `#'`:

```
$ clj-kondo --lint - <<< "(ns foo) (defn- foo []) foo (ns bar (:require [foo])) (var foo/foo)"
<stdin>:1:60: error: #'foo/foo is private
```

Looking into it now

**Comment by borkdude** [comment]
Should be fixed now. Can you test with the master version?

**Comment by mrkam2** [comment]
> Should be fixed now. Can you test with the master version?

Thanks for the quick fix! I verified that it got fixed.
